### ISSUE TITLE
--filter={gitmodified, gitstaged} is case-sensitive

### ISSUE THREAD

**Issue by simonsan** [squizlabs/PHP_CodeSniffer#3877]
## Describe the bug
![2023-08-19 03_45_26-Usage · squizlabs_PHP_CodeSniffer Wiki — Mozilla Firefox](https://github.com/squizlabs/PHP_CodeSniffer/assets/14062932/3629a7e5-afd7-4dee-b8e4-00253f939e26)

The documentation says, that a filter parameter can have one of the values `{gitmodified, gitstaged}`.

### Code sample
```php
phpcs -q --report=checkstyle --standard=PSR12 --filter=gitmodified --file-list=.phpcsinclude --tab-width=4 --parallel=10 | cs2pr
```

### To reproduce
1. run the command above
1. See error message displayed

```shell
PHP Fatal error:  Uncaught Error: Class "\PHP_CodeSniffer\Filters\gitmodified" not found in phar:///usr/local/bin/phpcs/src/Files/FileList.php:3
Stack trace:
#0 phar:///usr/local/bin/phpcs/src/Files/FileList.php(3): PHP_CodeSniffer\Files\FileList->addFile()
#1 phar:///usr/local/bin/phpcs/src/Runner.php(3): PHP_CodeSniffer\Files\FileList->__construct()
#2 phar:///usr/local/bin/phpcs/src/Runner.php(3): PHP_CodeSniffer\Runner->run()
#3 /usr/local/bin/phpcs(6): PHP_CodeSniffer\Runner->runPHPCS()
#4 {main}
  thrown in phar:///usr/local/bin/phpcs/src/Files/FileList.php on line 3
```

## Expected behaviour

The documentation states valid values. In this case: `{GitModified, GitStaged}`. Or the input value is converted to PascalCase.

## Please confirm:

- [X] I have searched the issue list and am not opening a duplicate issue.
- [X] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@simonsan Good catch! Would you like to submit a PR to fix this ?

**Comment by jrfnl** [comment]
FYI: For the Wiki, I've just fixed it: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Usage

**Comment by jrfnl** [comment]
FYI: the fix for this issue is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
